.circleci/delete.py

```diff
@@ -0,0 +1,198 @@
+        + "You are using {version}.".format(version=sys.version_info[0])
+    )
+
+transport = RequestsHTTPTransport(
+    url="https://{DASH_ENTERPRISE_HOST}/Manager/graphql".format(
+        DASH_ENTERPRISE_HOST=DASH_ENTERPRISE_HOST
+    ),
+    auth=(SERVICE_USERNAME, SERVICE_API_KEY),
+    use_json=True,
+    retries=5,
+)
+
+client = Client(transport=transport)
+
+PAGE = 0
+
+apps = []
+apps_result = []
+services = []
+
+print("Querying apps...\n")
+while len(apps_result) != 0 or PAGE == 0:
+    query = gql(
+        """
+        query($PAGE: Int) {
+        apps(page: $PAGE, allApps:true) {
+                apps {
+                    analytics {
+                        appname
+                        timestamps {
+                            created
+                            updated
+                        }
+                    }
+                    linkedServices {
+                        name
+                        serviceType
+                    }
+                }
+            }
+        }
+        """
+    )
+    params = {"PAGE": PAGE}
+    sleep(5)
+    api_call = client.execute(query, variable_values=params)
+    apps_result = api_call["apps"]["apps"]
+    apps.extend(apps_result)
+    print(f"  Page: {PAGE}")
+    PAGE = PAGE + 1
+print("\n  Apps: {apps}\n".format(apps=len(apps)))
+
+if len(apps) == 0:
+    print(
+        "No apps were found by user {username}.".format(username=SERVICE_USERNAME)
+        + "Check that {username} owns apps or has admin privileges.\n".format(
+            username=SERVICE_USERNAME
+        )
+    )
+    sys.exit()
+
+apps_name = []
+apps_updated = []
+apps_created = []
+services_name = []
+services_type = []
+apps_dict = dict()
+services_dict = dict()
+if len(apps) != 0:
+    print("Discovering linked databases associated with the apps...")
+    for i in enumerate(apps):
+        apps_name.append(apps[i]["analytics"]["appname"])
+        apps_created.append(apps[i]["analytics"]["timestamps"]["created"])
+        apps_updated.append(apps[i]["analytics"]["timestamps"]["updated"])
+        apps_dict.update(zip(apps_name, zip(apps_updated, apps_created)))
+        for j in range(len(apps[i]["linkedServices"])):
+            services_name.append(apps[i]["linkedServices"][j]["name"])
+            services_type.append(apps[i]["linkedServices"][j]["serviceType"])
+            services_dict.update(zip(services_name, zip(apps_name, services_type)))
+else:
+    print("No linked databases associated with the apps were discovered.")
+
+
+apps_filtered = dict()
+if len(apps) != 0:
+    print(
+        "Determining which apps haven't been updated or visited in "
+        + "{time} {period}...".format(
+            time=LAST_UPDATE.values(),
+            period=LAST_UPDATE.keys(),
+        )
+    )
+    for app_name, app_time in apps_dict.items():
+        if (
+            app_name.startswith("{prefix}".format(prefix=PREFIX))
+            and app_time[0] is None
```

Comment:
`app_time[0]` isn't descriptive which makes this code hard to read. Either:
1. Assign this to a variable above like `last_update_time = app_time[0]`
2. Or, add a comment here like `# app_time[0] is the last update time`